flake.nix

```diff
@@ -0,0 +1,337 @@
+          # devLibPath = pkgs.lib.makeLibraryPath [ pkgs.libressl.dev ];
+          # pname = "libressl";
+          # version = "3.9.2";
+          # includes = [ "tests/tlstest.sh" ];
+          # src = pkgs.fetchurl {
+          #   url = "mirror://openbsd/LibreSSL/${pname}-${version}.tar.gz";
+          #   hash = "sha256-ewMdrGSlnrbuMwT3/7ddrTOrjJ0nnIR/ksifuEYGj5c=";
+          # };
+          # nativeBuildInputs = _old.nativeBuildInputs ++ (with pkgs; [
+          #   pkg-config
+          # ]);
+
+          # Patched according to the previous versions:
+          # https://github.com/NixOS/nixpkgs/blob/nixos-24.05/pkgs/development/libraries/libressl/default.nix#L118
+          # For unknown reasons the newer versions are not patched this way (yet?)
+          patches = [
+            (pkgs.fetchpatch {
+              url = "https://github.com/libressl/portable/commit/86e4965d7f20c3a6afc41d95590c9f6abb4fe788.patch";
+              includes = [ "tests/tlstest.sh" ];
+              hash = "sha256-XmmKTvP6+QaWxyGFCX6/gDfME9GqBWSx4X8RH8QbDXA=";
+            })
+          ];
+
+          # NOTE: Due to name conflicts between OpenSSL and LibreSSL we need to resolve this manually.
+          postFixup =  pkgs.lib.concatLines [ 
+            ( _old.postFixup or "" )
+            ''
+            cp $dev/lib/pkgconfig/libcrypto.pc $dev/lib/pkgconfig/libresslcrypto.pc
+            sed --in-place --expression 's/-lcrypto/-lresslcrypto/' $dev/lib/pkgconfig/libresslcrypto.pc
+            ln -s $out/lib/libcrypto.so $out/lib/libresslcrypto.so
+            ''
+          ];
+
+        });
+        libresslShim = import ./nix/libresslshim.nix { pkgs = pkgs; libressl = libressl; };
+        boringsslShim = import ./nix/boringsslshim.nix { pkgs = pkgs; boringssl = patched_boringssl; };
+        mbedtlsShim = import ./nix/mbedtlsshim.nix { pkgs = pkgs; };
+        ippcryptoShim = import ./nix/ippcryptoshim.nix { pkgs = pkgs; ipp-crypto = customPkgs.ipp-crypto; };
+
+        overlays = [];
+        pkgs = import nixpkgs {
+          inherit system overlays;
+        };
+        customPkgs = import custom-nixpkgs {
+          inherit system overlays;
+        };
+        buildECTesterStandalone = { opensslVersion, opensslHash }: (
+          let
+            patched_openssl = pkgs.openssl.overrideAttrs (_old: rec {
+              version = opensslVersion;
+              pname = "openssl";
+              src = pkgs.fetchurl {
+                url = "https://www.openssl.org/source/openssl-${version}.tar.gz";
+                hash = opensslHash;
+              };
+              # FIXME Removing patches might cause unwanted things.
+              patches = [];
+            });
+
+            # devLibPath = pkgs.lib.makeLibraryPath [ pkgs.libressl.dev ];
+            # libressl = pkgs.libressl.overrideAttrs (_old: {
+            #   fixupPhase = ''
+            #     cp ${devLibPath}/openssl.pc ${devLibPath}/libressl.pc
+            #   '';
+            # });
```

Comment:
This can be removed right?
```suggestion
```

Reply:
As said below, there is still work to be done. Cleaning up and polishing is still down the pipeline.